test/framework/http.go

```diff
@@ -72,7 +74,27 @@ func WaitForSuccessfulHTTPRequest(namespace string, requestInfo HTTPRequestInfo,
 
 // ExecuteHTTPRequest executes an HTTP request
 func ExecuteHTTPRequest(namespace string, requestInfo HTTPRequestInfo) (*http.Response, error) {
-	return ExecuteHTTPRequestC(&http.Client{}, namespace, requestInfo)
+	// Setup a retry in case the first time it did not work
+	retry := 0
+	var resp *http.Response
+	var err error
+	for retry < config.GetHTTPRetryNumber() {
+		resp, err := ExecuteHTTPRequestC(&http.Client{}, namespace, requestInfo)
+		if err == nil && checkHTTPResponseSuccessful(namespace, resp) {
+			return resp, err
+		} else if resp.StatusCode == 500 {
```

Comment:
Looking in ExecuteHTTPRequestC, it seems that the response could return null in the response with an error happened.   (Scenario when err is not null and resp is null). So I guess, we should either do the else if err != nil first, or check whether resp is not nil here.

Reply:
good catch. will update